### PR TITLE
Fix: in-app browser is blank when open marketing consent dialog's privacy web page

### PR DESCRIPTION
**Pull request by billypchan**
### Issues
After https://github.com/wireapp/wire-ios/pull/2205 is merged, In-app browser is blank when open marketing consent dialog's privacy web page.

### Causes

SFSafariViewController cannot render correctly if it is not presented with the same window as the current view controller.

### Solutions

Present BrowserViewController from the same view controller instead of notificationsWindow.rootViewController.

## Notes

After testing, I found that BrowserViewController does not call completion closure. Rewrite BrowserViewController in a simpler way to fix it.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationTeam/UIAlertController+Newsletter.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ extension UIAlertController {
     /// email regisration work flow: newsletter subscription dialog appears after conversation list is displayed.)
     static var newsletterSubscriptionDialogWasDisplayed = false
 
-    static func showNewsletterSubscriptionDialog(completionHandler: @escaping (Bool) -> Void) {
+    static func showNewsletterSubscriptionDialog(presentViewController:UIViewController, completionHandler: @escaping (Bool) -> Void) {
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationTeam/UIAlertController+Newsletter.swift]: Would be slightly more clear I think to create the signature like showNewsletterSubscriptionDialog(over viewController: UIViewController, ...`

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationTeam/Flow/TeamCreationFlowController.swift [hunk 2]
```diff
@@ -123,9 +123,17 @@ extension TeamCreationFlowController {
         }
     }
 
+    fileprivate func showMarketingConsentDialog(presentViewController: UIViewController) {
+        UIAlertController.newsletterSubscriptionDialogWasDisplayed = false
+        UIAlertController.showNewsletterSubscriptionDialogIfNeeded(presentViewController: presentViewController) { [weak self] marketingconsent in
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationTeam/Flow/TeamCreationFlowController.swift]: `marketingconsent` → `marketingConsent`

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationTeam/Flow/TeamCreationFlowController.swift [hunk 3]
```diff
@@ -123,9 +123,17 @@ extension TeamCreationFlowController {
...
+            self?.marketingConsent = marketingconsent
+        }
+    }
+
     fileprivate func pushController(for state: TeamCreationState) {
 
         var stepDescription: TeamCreationStepDescription?
+        var needToShowMarketingConsentDialog = false
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationTeam/Flow/TeamCreationFlowController.swift]: I'd rename this `needsToShowMarketingConsentDialog`

File: Wire-iOS/Sources/UserInterface/Components/Controllers/BrowserViewController.swift [hunk 4]
```diff
@@ -21,32 +21,13 @@ import SafariServices
...
-
-    }
-
     @objc var completion: (() -> Void)? {
-        get {
-            return (delegate as? BrowserDelegate)?.completion
-        }
-        set {
-            guard let block = newValue else {
+        didSet {
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Components/Controllers/BrowserViewController.swift]: Could also override `viewDidLoad` and just set the delegate once in there. Then the whole `didSet` logic could go.

**Pull request merged**